File: dp.py
class Solution:
    def maximumProfit(self, n: int, prices: [int]) -> int:
        dp = [0] * (n+1)
        dp[1] = prices[0]
        for i in range(2, n+1):
            _max = 0
            for j in range(1, i+1):
                _max = max(_max, prices[j-1] + dp[i-j])
            dp[i] = _max
        return dp[n]


A = [1, 3, 4, 5, 7, 9, 10, 11]
n = 8

solution = Solution()
print(solution.maximumProfit(8, A))

File: dp.cpp
#include <iostream>
#include <vector>
using namespace std;

int maximumProfit(int n, vector<int> &prices) {
    // add your logic here
	vector<int> dp(n+1, 0);
	dp[1] = prices[0];
	for(int i = 2; i <= n; i++){
		int _max = 0;
		for(int j = 1; j <= i; j++){
			if(_max < prices[j-1] + dp[i-j])
				_max = prices[j-1] + dp[i-j];
		}
		dp[i] = _max;
	}
	return dp[n];
}
